sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c

```diff
@@ -0,0 +1,487 @@
+  assert_int_equal(mqtt_sub_topic_buf[az_span_length(mqtt_sub_topic)], 0xFF);
+}
+
+static void test_az_iot_hub_client_methods_response_publish_topic_get_INSUFFICIENT_BUFFER_for_prefix_fail(void** state)
+{
+  (void)state;
+
+  uint8_t mqtt_sub_topic_buf[10];
+  memset(mqtt_sub_topic_buf, 0xFF, _az_COUNTOF(mqtt_sub_topic_buf));
+  az_span mqtt_sub_topic = az_span_init(mqtt_sub_topic_buf, 0, _az_COUNTOF(mqtt_sub_topic_buf));
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span request_id = AZ_SPAN_LITERAL_FROM_STR("2");
+  uint16_t status = 202;
+
+  assert_true(az_iot_hub_client_methods_response_publish_topic_get(&client, request_id, status, mqtt_sub_topic, &mqtt_sub_topic) == AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY);
+}
+
+static void test_az_iot_hub_client_methods_response_publish_topic_get_INSUFFICIENT_BUFFER_for_status_fail(void** state)
+{
+  (void)state;
+
+  uint8_t mqtt_sub_topic_buf[21]; // Enough for "$iothub/methods/res/2"
+  memset(mqtt_sub_topic_buf, 0xFF, _az_COUNTOF(mqtt_sub_topic_buf));
+  az_span mqtt_sub_topic = az_span_init(mqtt_sub_topic_buf, 0, _az_COUNTOF(mqtt_sub_topic_buf));
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span request_id = AZ_SPAN_LITERAL_FROM_STR("2");
+  uint16_t status = 202;
+
+  assert_true(az_iot_hub_client_methods_response_publish_topic_get(&client, request_id, status, mqtt_sub_topic, &mqtt_sub_topic) == AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY);
+}
+
+static void test_az_iot_hub_client_methods_response_publish_topic_get_INSUFFICIENT_BUFFER_for_reqid_fail(void** state)
+{
+  (void)state;
+
+  uint8_t mqtt_sub_topic_buf[24]; // Enough for "$iothub/methods/res/202/"
+  memset(mqtt_sub_topic_buf, 0xFF, _az_COUNTOF(mqtt_sub_topic_buf));
+  az_span mqtt_sub_topic = az_span_init(mqtt_sub_topic_buf, 0, _az_COUNTOF(mqtt_sub_topic_buf));
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span request_id = AZ_SPAN_LITERAL_FROM_STR("2");
+  uint16_t status = 202;
+
+  assert_true(az_iot_hub_client_methods_response_publish_topic_get(&client, request_id, status, mqtt_sub_topic, &mqtt_sub_topic) == AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY);
+}
+
+static void test_az_iot_hub_client_methods_received_topic_parse_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
```

Comment:
nit: `hub_client_init` accepts `NULL` as a param for options. It will then call `az_iot_hub_client_options_default()` for you.

Reply:
That's true. I should simplify that. Thanks.